projects/com.oracle.truffle.llvm.types/src/com/oracle/truffle/llvm/types/LLVMFunctionDescriptor.java

```diff
@@ -45,6 +46,7 @@
 import com.oracle.truffle.api.nodes.RootNode;
 import com.oracle.truffle.llvm.types.memory.LLVMStack;
 
+@ValueType
 public final class LLVMFunctionDescriptor implements TruffleObject, Comparable<LLVMFunctionDescriptor> {
```

Comment:
I doubt that this is correct here, because this type is potentially stored in Maps and passed around to unsuspecting foreign languages. This can be a problem since ValueType breaks some of the most basic contracts about the behavior of Java objects.